src/main/java/com/google/impactdashboard/database_manager/FakeDatabase.java

```diff
@@ -0,0 +1,203 @@
+        });
+
+        put(ProjectIdentification.create("project-1", "project-id-1", 
+          Long.parseLong("123456789123")), bindingsProject1);
+
+        HashMap<Long, Integer> bindingsProject2 = new HashMap<Long, Integer>();
+        AtomicReference<Long> date2 = new AtomicReference<Long>(Long.parseLong("1590883200000"));
+
+        Arrays.asList(500, 500, 750, 750, 750, 750, 750, 1000, 1000, 
+        1000, 500, 500, 500, 600, 600, 600, 600, 300, 300, 1000, 
+        1000, 1000, 1100, 1100, 1000, 1000, 500, 500, 500, 500)
+        .forEach(numberBindings -> {
+          bindingsProject2.put(date2.get(), numberBindings);
+          date2.set(date2.get() + MILLISECONDS_ONE_DAY);
+        });
+
+        put(ProjectIdentification.create("project-2", "project-id-2", 
+          Long.parseLong("234567890123")), bindingsProject2);
+      }
+  };
+
+  /** 
+   * Returns the identifying information of the projects that appear in the 
+   * IAM Bindings table representation. 
+   */
+  public static List<ProjectIdentification> listProjects() {
+    List<ProjectIdentification> projects = new ArrayList<ProjectIdentification>();
+    iamBindings.forEach((project, data) -> projects.add(project));
+    return projects;
+  }
+
+  /** Retrieves the bindings data associated with {@code projectId} in the bindings table. */
+  public static Map<Long, Integer> getDatesToBindingsForProject(String projectId) {
+    AtomicReference<Map<Long, Integer>> dailyBindings = 
+      new AtomicReference<Map<Long, Integer>>();
+    iamBindings.forEach((project, data) -> {
+      if (project.getProjectId().equals(projectId)) {
+        dailyBindings.set(data);
+      }
+    });
+    return dailyBindings.get();
+  }
+
+  /** 
+   * Returns the average bindings recorded in the bindings table for a given 
+   * project, or 0 if this project does not appear in the bindings table. 
+   */
+  public static double getAvgBindingsForProject(String projectId) {
+    Map<Long, Integer> dailyBindings = getDatesToBindingsForProject(projectId);
+
+    if (dailyBindings.size() == 0) {
+      return 0;
+    }
+
+    AtomicReference<Double> averageBindings = new AtomicReference<Double>();
+    dailyBindings.forEach((date, bindings) -> {
+      averageBindings.set(averageBindings.get() + bindings);
+    });
+    return averageBindings.get() / dailyBindings.size();
+  }
+
+  /** 
+   * Returns a map of entries in the recommendations table associated with 
+   * {@code projectId}. 
+   */
+  public static Map<Long, Recommendation> getDatesToRecommendationsForProject(
+    String projectId) {
+    return recommendations.entrySet().stream()
+      .filter(mapElement -> 
+      ((Recommendation) mapElement.getValue()).getProjectId().equals(projectId))
+      .collect(Collectors.toMap(
+        mapElement -> mapElement.getKey(), mapElement -> mapElement.getValue()));
+  }
+
+  /** Adds {@code newRecommendations} to existing table of recommendations. */
```

Comment:
I don't know if this would be better but maybe do @param newRecommendation in the javadoc so if someone is only looking at the java docs it will specify what the parameters are.

Reply:
Ah, right, makes sense. I'll go add in some params.